pkg/storage/blobstore/s3/s3.go

```diff
@@ -73,3 +88,28 @@ func NewStorage(c map[string]any) (*Storage, error) {
 
 	return q, nil
 }
+
+func NewStorageWithCreds(accessKeyID, secretAccessKey, bucket, region string) (*Storage, error) {
```

Comment:
This function should just call `NewStorage()` with the parameters passed in. Do not call `LoadDefaultConfig()`, that is only intended to be used if someone runs the program without any config file at all.